File: src/06-guessing-game/rust/src/main.rs
// I followed along
// Programming a Guessing Game in Rust!
// https://www.youtube.com/watch?v=H0xBSbnQYds
//
// Some modifications were needed:
// * parse
// * gen_range

use colored::*;
use rand::Rng;
use std::io;

fn main() {
    println!("Guess the number!");

    let secret_number = rand::thread_rng().gen_range(1..101);

    println!("The secret number is {secret_number}");

    loop {
        println!("Please input your guess");

        let mut guess = String::new();

        io::stdin()
            .read_line(&mut guess)
            .expect("Failed to read line");

        let guess = match guess.trim().parse::<u32>() {
            Ok(num) => num,
            Err(_) => continue,
        };

        println!("You guessed: {guess}");

        match guess.cmp(&secret_number) {
            std::cmp::Ordering::Less => println!("{}", "Too small!".red()),
            std::cmp::Ordering::Greater => println!("{}", "Too big!".red()),
            std::cmp::Ordering::Equal => {
                println!("{}", "You win!".green());
                break;
            }
        }
    }
}

File: src/06-guessing-game/typescript/src/main.ts
/*!
 * Polyglot Programming
 *
 * Copyright (c) 2024- Adam Burucs
 *
 * MIT Licensed
 */

import * as readlineSync from "readline-sync";
import chalk from "chalk";

function getRandomInt(min: number, max: number): number {
  return Math.floor(Math.random() * (max - min)) + min;
}

function main() {
  console.log("Guess the number!");

  const secretNumber = getRandomInt(1, 101);
  console.log(`The secret number is ${secretNumber}`);

  while (true) {
    const guessInput = readlineSync.question("Please input your guess: ");

    const guess = parseInt(guessInput.trim(), 10);
    if (isNaN(guess)) {
      continue;
    }

    console.log(`You guessed: ${guess}`);

    if (guess < secretNumber) {
      console.log(chalk.red("Too small!"));
    } else if (guess > secretNumber) {
      console.log(chalk.red("Too big!"));
    } else {
      console.log(chalk.green("You win!"));
      break;
    }
  }
}

main();
